src/groupeddataframe/fastaggregates.jl

```diff
@@ -193,6 +194,12 @@ function groupreduce!(res::AbstractVector, f, op, condf, adjust, checkempty::Boo
             end
         end
     end
+    # Eliminate the argument-type diversity that we no longer need
+    return _groupreduce!(res, counts, n, op === min || op === max, adjust, checkempty, incol)
+end
+
+function _groupreduce!(res::AbstractVector, counts::Union{Nothing,Vector{Int}}, n::Int, opminormax::Bool, adjust, checkempty::Bool,
+                      incol::AbstractVector)
     if adjust !== nothing
         res .= adjust.(res, counts)
     end
```

Comment:
Putting this block and the next one in this method doesn't sound great, as `adjust` and `counts` are tightly linked to `op` (they are used for `mean` basically). If you keep them in the parent method this one won't have to be specialized on the particular operation.